tests/pytorch/distributed/run_fsdp2_model.py

```diff
@@ -157,19 +166,34 @@ def _train(args):
     fully_shard(model, mesh=mesh)
     restore_custom_attrs(model, custom_attrs)
 
+    if LOCAL_RANK == 0:
+        print(f"Rank {LOCAL_RANK}: FSDP applied, now materializing sharded parameters...")
+        print("Memory after FSDP:", torch.cuda.memory_allocated(device)/1e6, "MB")
+
+    # After FSDP has been applied, materialize and initialize the sharded parameters
+    # TransformerEngine's reset_parameters() now properly handles DTensors and FP8 initialization
+    for module in model.modules():
+        if hasattr(module, 'reset_parameters'):
+            module.reset_parameters()
+
+    if LOCAL_RANK == 0:
+        print(f"Rank {LOCAL_RANK}: Sharded parameters materialized and initialized.")
+        print("Memory after materialization:", torch.cuda.memory_allocated(device)/1e6, "MB")
+
     optimizer = optim.Adam(model.parameters(), lr=1e-3)
 
     for iteration in range(args.iter):
         # Zero the parameter gradients
         optimizer.zero_grad()
         input_data = torch.randn(args.batch_size, args.input_size).to(device)
-        output = model(input_data)
+        with te.fp8_autocast(enabled=True, fp8_recipe=fp8_recipe):
```

Comment:
We've deprecated `fp8_autocast` in https://github.com/NVIDIA/TransformerEngine/pull/2256 since we also support FP4:
```suggestion
        with te.autocast(enabled=True, recipe=fp8_recipe):
```

Reply:
Fixed